dipy/nn/cnn_1denoiser/cnn_1denoiser.py

```diff
@@ -0,0 +1,283 @@
+activation function was used in both convolutional layers. There was 1 max-pooling layer that had a kernel
+size of 2 with stride 2. In the dense layer, the extracted features of the high-noise image were mapped to
+the low-noise reference image.
+
+reference:
+---------
+Denoising diffusion weighted imaging data using convolutional neural networks
+Hu Cheng, Sophia Vinci-Booher, Jian Wang, Bradley Caron, Qiuting Wen, Sharlene Newman, Franco Pestilli
+bioRxiv 2022.01.17.476708; doi: https://doi.org/10.1101/2022.01.17.476708
+
+Contributors:
+------------
+-Rahul Suresh Ubale :implemented the cnn_1denoiser model and tested the code.
+-Hu Cheng :provided guidance and support for the project.
+-Eleftherios Garyfallidis :provided guidance and support for the project.
+'''
+class cnn_1denoiser:
+    def __init__(self, sig_length, optimizer='adam', loss='mean_squared_error', metrics=('accuracy',),
+         loss_weights=None): 
+        '''
+        Parameters
+        ----------
+        sig_length : Int
+            Shape of data(Time) 
+        optimizer : str, optional  
+            Adam (Default) optimization algorithm name
+        loss : str, optional ,mean_squared_error(Default)[fixed]
+            Select loss function for measuring accuracy.
+            Computes the mean of squares of errors between labels and predictions
+        metrics : Funtion name 
+            accuracy (Default) ,judge the performance of your model
+        optimizer : str, optional
+            Select optimizer. Default adam.
+        loss : floats
+            Optional list or dictionary specifying scalar coefficients to weight.
+            
+        '''
+    # Layer 1 - Convolutional Layer(16*1*16) + ReLU activation
+    #           Pooling
+    # Layer 2 - Convolutional Layer(8*16*32) + ReLU activation
+    #           Pooling
+    #           Flatten
+    #           Dense 
+        input_layer = Input(shape=(sig_length, 1))
+        x = Conv1D(filters=16, kernel_size=16, kernel_initializer='Orthogonal',
+                padding='same', name=f'Conv1')(input_layer)
+        x = Activation('relu', name=f'ReLU1')(x)
+        max_pool_1d = tf.keras.layers.MaxPooling1D(pool_size=2, strides=2, padding='valid')
+        pool1 = max_pool_1d(x)
+        x = Conv1D(filters=32, kernel_size=8, padding='same', name=f'Conv2')(pool1)
+        x = Activation('relu', name=f'ReLU2')(x)
+        pool2 = max_pool_1d(x)
+        pool2_flat = tf.keras.layers.Flatten()(pool2)
+        logits = tf.keras.layers.Dense(units=sig_length, activation='relu')(pool2_flat)
+        model = Model(inputs=input_layer, outputs=logits)
+        model.compile(optimizer=optimizer, loss=loss, metrics=metrics,loss_weights=loss_weights)
+        self.model = model
+
+    def compile(self, optimizer='adam', loss=None, metrics=None,loss_weights=None):
+        self.model.compile(optimizer=optimizer, loss=loss, metrics=metrics,
```

Comment:
Please add a docstring. In particular what is the default for "loss" (what happens when it's set to 'None'?)